src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -2261,11 +2261,9 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
     }
 
 #if defined(FEATURE_MASKED_HW_INTRINSICS) && defined(TARGET_ARM64)
-    auto convertToMaskIfNeeded = [&](GenTree*& op) {
-        if (!varTypeIsMask(op))
-        {
-            op = gtNewSimdCvtVectorToMaskNode(TYP_MASK, op, simdBaseJitType, simdSize);
-        }
+    auto convertToMask = [&](GenTree*& op) {
+        assert(varTypeIsSIMD(op));
+        op = gtNewSimdCvtVectorToMaskNode(TYP_MASK, op, simdBaseJitType, simdSize);
     };
```

Comment:
We can get rid of `convertToMask` entirely and just use `gtNewSimdCvtVectorToMaskNode` directly. That is more clear, removed unnecessary overhead, and doesn't lose any functionality (an `assert(varTypeIsSIMD(op)` already exists in `gtNewSimdCvtVectorToMaskNode`).

Reply:
I considered getting rid of `convertToMask()` but I didn't realise the assert was inside `gtNewSimdCvtVectorToMaskNode()`. Will fix.

Reply:
`convertToMask()` removed.